src/realtime_metrics/main.py

```diff
@@ -6,17 +6,168 @@
 from sqlalchemy import create_engine, inspect
 from sqlalchemy.orm import sessionmaker
 
-import pandas as pd
+import numpy
+
+def run_analysis():
+    # get all stop time updates and run compute the different metrics
+    tripUpdates = session.query(TripUpdate).group_by(TripUpdate.trip_id).all()
+
+    stop_time_updates: list[tuple[TripUpdate, StopTimeUpdate]] = []
+
+    for tripUpdate in tripUpdates:
+        trip_stop_time_updates = session.query(StopTimeUpdate, TripUpdate).join(TripUpdate).filter(TripUpdate.trip_id == tripUpdate.trip_id).filter(StopTimeUpdate.arrival_time > 0).all()
+        for trip_stop_time_update in trip_stop_time_updates:
+            stop_time_updates.append((trip_stop_time_update.TripUpdate, trip_stop_time_update.StopTimeUpdate))
+
+            # check dict
+            stop_time_update: StopTimeUpdate = trip_stop_time_update.StopTimeUpdate
+            key = (tripUpdate.route_id, tripUpdate.trip_id, stop_time_update.stop_id)
+            if key in dictionary.keys():
+                if tripUpdate.timestamp.timestamp() > dictionary[key][0]:
+                    dictionary[key] = (tripUpdate.timestamp.timestamp(), stop_time_update)
+            else:
+                dictionary[key] = (tripUpdate.timestamp.timestamp(), stop_time_update)
+
+    # compute accuracy metric
+    logger.info("MSE accuracy: ")
+    mse_accuracy(stop_time_updates=stop_time_updates)
+    logger.info("ETA accuracy:")
+    eta_accuracy(stop_time_updates=stop_time_updates)
+
+
+def mse_accuracy(stop_time_updates: list[tuple[TripUpdate, StopTimeUpdate]]):
+    # computes the accuracy of the given stop time updates using mean squared error
+
+    if len(stop_time_updates) <= 0:
+        logger.info("No samples provided!")
+        return
+
+    # compute prediction error for each stop time update
+    samples = []
+    for update in stop_time_updates:
+        trip_update = update[0]
+        stop_time_update = update[1]
+        actual_delay = get_actual_delay(trip_update=trip_update, stop_time_update=stop_time_update)
+        predicted_delay = stop_time_update.arrival_delay
+        prediction_error = actual_delay - predicted_delay
+        samples.append(prediction_error)
+
+    # compute MSE
+    sum = 0
+    for entry in samples:
+        sum += entry * entry
+    mean_squared_error = sum / len(samples)
+
+    # print MSE
+    logger.info("MSE: %s", mean_squared_error)
```

Comment:
Return the value, no logger info here, but instead log it where you call the function
```suggestion
    return mean_squared_error
```